voyager.js

```diff
@@ -45,31 +45,64 @@ new BasicTracerProvider().register()
 Zinnia.activity.info('Voyager benchmarking started')
 
 export async function runSaturnBenchmarkInterval() {
-    const random = Math.random()
-    if (random <= prodOpts.sampleRate) {
-        console.log('Running prod benchmark...')
-        try {
-            await runBenchmark(prodSaturn)
-            console.log('Prod benchmark successful!')
-        } catch (err) {
-            console.error(err)
-        }
-        Zinnia.jobCompleted()
+    let runningOnSaturnNode = false
+    try {
+        runningOnSaturnNode = await isRunningOnSaturnNode()
+    } catch (err) {
+        console.error(
+            'Could not check if running on Saturn node',
+            { cause: err }
+        )
     }
-    if (random <= testOpts.sampleRate) {
-        console.log('Running test benchmark...')
-        try {
-            await runBenchmark(testSaturn)
-            console.log('Test benchmark successful!')
-        } catch (err) {
-            console.error(err)
+    if (runningOnSaturnNode) {
+        console.log('Running on Saturn host, skipping benchmark')
+    } else {
+        const random = Math.random()
+        if (random <= prodOpts.sampleRate) {
+            console.log('Running prod benchmark...')
+            try {
+                await runBenchmark(prodSaturn)
+                console.log('Prod benchmark successful!')
+            } catch (err) {
+                console.error(err)
+            }
+            Zinnia.jobCompleted()
+        }
+        if (random <= testOpts.sampleRate) {
+            console.log('Running test benchmark...')
+            try {
+                await runBenchmark(testSaturn)
+                console.log('Test benchmark successful!')
+            } catch (err) {
+                console.error(err)
+            }
+            Zinnia.jobCompleted()
         }
-        Zinnia.jobCompleted()
     }
     console.log('Sleeping for 60s...')
     setTimeout(runSaturnBenchmarkInterval, 1000 * 60)
 }
 
+async function getPublicIPv4Address () {
+    // TODO Replace with Voyager API once available
+    const res = await fetch(`https://api.filspark.com/inspect-request`)
+    const { cloudfareAddr: ip } = await res.json()
```

Comment:
should this be `cloudflareAddr`? (missing an `l`)

Reply:
oh golly I just went and looked and no, the `l` is missing in the response, lol!

Reply:
hahah good catch I wasn't aware!

Reply:
https://github.com/filecoin-station/spark-api/pull/232